libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/compilerRunner/GradleKotlinCompilerRunner.kt

```diff
@@ -367,25 +367,21 @@ internal class GradleCompilerRunner(private val project: Project) : KotlinCompil
             val jarToModule = HashMap<File, IncrementalModuleEntry>()
 
             for (project in gradle.rootProject.allprojects) {
-                for (task in project.tasks) {
-                    when (task) {
-                        is AbstractKotlinCompile<*> -> {
-                            val module = IncrementalModuleEntry(project.path, task.moduleName, project.buildDir, task.buildHistoryFile)
-                            dirToModule[task.destinationDir] = module
-                            task.javaOutputDir?.let { dirToModule[it] = module }
-                            nameToModules.getOrPut(module.name) { HashSet() }.add(module)
-
-                            if (task is Kotlin2JsCompile) {
-                                jarForSourceSet(project, task.sourceSetName)?.let {
-                                    jarToModule[it] = module
-                                }
-                            }
-                        }
-                        is InspectClassesForMultiModuleIC -> {
-                            jarToClassListFile[File(task.archivePath)] = task.classesListFile
+                project.tasks.withType(AbstractKotlinCompile::class.java).forEach { task ->
```

Comment:
If you want to use the new lazy configuration API, this needs to be `configureEach`.
Also, `forEach` won't configure new `AbstractKotlinCompile` tasks that are added later in the build. Is this intentional?